### PR TITLE
Grocksdb and docker

### PR DESCRIPTION
**Pull request by faddat**
This PR introduces grocksdb and a docker container that has rocksdb v7.x in it

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
     # rocksdb, which are needed to build and test those modules.
     # To update the container image, see docker.yml.
     runs-on: ubuntu-latest
-    container: tendermintdev/docker-tm-db-testing
+    container: ghcr.io/faddat/cosmos
```
Comment (marbar3778) [inline .github/workflows/ci.yml:21]: reasoning?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -1,19 +1,42 @@
...
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
 jobs:
   golangci:
-    # We need to run the linter on the same image we use for building, since it
-    # needs the C libraries installed for the dependencies to typecheck.
+    name: lint
     runs-on: ubuntu-latest
-    container: tendermintdev/docker-tm-db-testing
+    container: ghcr.io/faddat/cosmos
```
Comment (marbar3778) [inline .github/workflows/lint.yml:13]: ditto

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -1,19 +1,42 @@
...
+
+          # Optional: if set to true then the all caching functionality will be complete disabled,
+          #           takes precedence over all other caching options.
+          # skip-cache: true
+
+          # Optional: if set to true then the action don't cache or restore ~/go/pkg.
+          # skip-pkg-cache: true
+
+          # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
+          # skip-build-cache: true
```
Comment (marbar3778) [inline .github/workflows/lint.yml:42]:
```suggestion
```

**Review by marbar3778 — COMMENTED** [review]
Could you add a changelog entry and some reasoning as to why this change is needed

**Comment by faddat** [comment]
You're right it definitely needs docs.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (faddat) [reply .github/workflows/ci.yml:21]: reasoning:  when we bump rocksdb, we also need to use a different docker image.  The alternative is to upgrade the current docker image.

File: .github/workflows/lint.yml (hunk 2 above)
Comment (faddat) [reply .github/workflows/lint.yml:13]: reasoning: when we bump rocksdb, we also need to use a different docker image. The alternative is to upgrade the current docker image.

**Pull request closed without merging**